### PR TITLE
AP_RangeFinder: change error and ok status defines to LeddarOne_Status enum

### PR DESCRIPTION
**Pull request by ShingoMatsuura**
change error and ok status defines to LeddarOne_Status enum. And change send_request, parse_response function's return value from int8_t to LeddarOne_Status.

### CODE REVIEW

File: libraries/AP_RangeFinder/AP_RangeFinder_LeddarOne.cpp [hunk 1]
```diff
@@ -67,17 +67,15 @@ bool AP_RangeFinder_LeddarOne::get_reading(uint16_t &reading_cm)
         }
     }
 
-    // parse a response message, set detections and sum_distance
+    // parse a response message, set number_detectins, detections and sum_distance
```
Comment (lucasdemarchi) [inline libraries/AP_RangeFinder/AP_RangeFinder_LeddarOne.cpp]: number_detections

Comment (ShingoMatsuura) [inline libraries/AP_RangeFinder/AP_RangeFinder_LeddarOne.cpp]: Thank you for reviewing. Im going to modify the spelling.

**Comment by ShingoMatsuura** [comment]
@muramura , Thank you for reviewing. I'm going to change C type cast to static_cast.

**Comment by ShingoMatsuura** [comment]
@magicrub, Thank you for reviewing and comment. I am going to rework.

**Comment by ShingoMatsuura** [comment]
Thank yon for merg! @rmackay9,@lucasdemarchi.

**Pull request merged**